library/printoptions.ml

```diff
@@ -0,0 +1,108 @@
+(* record of Printing options *)
+
+type t = {
+    printing_allow_match_default_clause : bool;
+    printing_coercions : bool;
+    printing_compact_contexts : bool;
+    printing_existential_instances : bool;
+    printing_factorizable_match_patterns : bool;
+    printing_implicit: bool;
+    printing_implicit_defensive : bool;
+    printing_matching : bool;
+    printing_notations : bool;
+    printing_primitive_projection_compatibility : bool;
+    printing_primitive_projection_parameters : bool;
+    printing_projections : bool;
+    printing_records : bool;
+    printing_synth : bool;
+    printing_universes : bool;
+    printing_wildcard : bool;
+  }
+
+let default_options : t = {
+    printing_allow_match_default_clause = true;
+    printing_coercions = false;
+    printing_compact_contexts = false;
+    printing_existential_instances = false;
+    printing_factorizable_match_patterns = true;
+    printing_implicit = false;
+    printing_implicit_defensive = true;
+    printing_matching = true;
+    printing_notations = true;
+    printing_primitive_projection_compatibility = false;
+    printing_primitive_projection_parameters = false;
+    printing_projections = false;
+    printing_records = true;
+    printing_synth = true;
+    printing_universes = false;
+    printing_wildcard = true;
+  }
+
+let all_options_enabled : t = {
+    printing_allow_match_default_clause = false;
+    printing_coercions = true;
+    printing_compact_contexts = true;
+    printing_existential_instances = true;
+    printing_factorizable_match_patterns = false;
+    printing_implicit = true;
+    printing_implicit_defensive = true;
+    printing_matching = false;
+    printing_notations = false;
+    printing_primitive_projection_compatibility = true;
+    printing_primitive_projection_parameters = true;
+    printing_projections = false;
+    printing_records = true;
```

Comment:
Not sure about this one, but I guess it's most like the previous behaviour?

Reply:
The relevant code in `constrextern.ml` is: 
```
            if !Flags.raw_print then raise Exit;
            let cstrsp = match ref with ConstructRef c -> c | _ -> raise Not_found in
            let struc = Recordops.lookup_structure (fst cstrsp) in
            if PrintingRecord.active (fst cstrsp) then
                  ()
            else if PrintingConstructor.active (fst cstrsp) then
                raise Exit
            else if not !record_print then
                raise Exit;
            [ record printing code, I guess is here ]
```
So looking at this now, I'd say maybe printing records should be false in the all-enabled record. Do you agree?

Reply:
Indeed I missed that raw_print when I looked since it's earlier than the record_print.

Reply:
In https://github.com/coq/coq/pull/6698/commits/ebb0ec352b1e96c531cc21ab00f2d8fd8f62491b, I changed `Printing All` to print records, because that reveals more detail than the constructor presentation. That's not the current behavior, though.

Reply:
And in 35edb24, I reverted to not printing records in `Set Printing All`. Record types were only  printed as `Record` in 8.7.1 with `Unset Printing All`. It makes sense to print record instances to print as records in the same situation, and both types/instances are controlled by `Printing Records`.